File: app/src/main/java/com/amperas17/mangaedenapp/model/Resource.java
package com.amperas17.mangaedenapp.model;


public class Resource<T> {

    private T data;
    private Throwable throwable;

    public Resource(T data) {
        this.data = data;
        this.throwable = null;
    }

    public Resource(Throwable error) {
        this.throwable = error;
        this.data = null;
    }

    public T getData() {
        return data;
    }

    public Throwable getThrowable() {
        return throwable;
    }

}

File: app/src/main/java/com/amperas17/mangaedenapp/model/Resource.kt
package com.amperas17.mangaedenapp.model


class Resource<T> {

    var data: T? = null
        private set
    var throwable: Throwable? = null
        private set

    constructor(data: T) {
        this.data = data
        this.throwable = null
    }

    constructor(error: Throwable) {
        this.throwable = error
        this.data = null
    }
}
